### PR TITLE
incorporating relevant changes from led_strip v3.0.0 (BSP-574)

### PR DESCRIPTION
**Pull request by rishi9699**
# ESP-BSP Pull Request checklist

> Note: For new BSPs create a PR with this [link](https://github.com/espressif/esp-bsp/compare/main...my-branch?quick_pull=1&template=pr_template_bsp.md).
- [ ] Version of modified component bumped
- [ ] CI passing

# Change description
Recently a new version (v3.0.0) of led_strip was pushed. This changed the `led_strip_config_t` struct in the 

> led_strip/include/led_strip_types.h

 file.

Please check the latest change of led_strip repository [here](https://github.com/espressif/idf-extra-components/commit/31585c8c97b5a403e1550270fb9fad58879cc2dd)

Closes https://github.com/espressif/esp-bsp/issues/420

### CODE REVIEW

**Comment by igrr** [comment]
cc @espzav @tore-espressif

I think we also need to add a version constraint on led_strip somewhere, or possibly have ifdefs for led_strip v2 and v3.

File: bsp/esp_bsp_generic/src/esp_bsp_generic.c [hunk 1]
```diff
@@ -194,8 +194,8 @@ static led_indicator_gpio_config_t bsp_leds_gpio_config[] = {
 static const led_strip_config_t bsp_leds_rgb_strip_config = {
     .strip_gpio_num = CONFIG_BSP_LED_RGB_GPIO,   // The GPIO that connected to the LED strip's data line
     .max_leds = BSP_LED_NUM,                  // The number of LEDs in the strip,
-    .led_pixel_format = LED_PIXEL_FORMAT_GRB, // Pixel format of your LED strip
     .led_model = LED_MODEL_WS2812,            // LED strip model
+    .color_component_format = LED_STRIP_COLOR_COMPONENT_FMT_GRB,  // LED color component format
```
Comment (suda-morris) [inline bsp/esp_bsp_generic/src/esp_bsp_generic.c:198]:
```suggestion
```

Comment (suda-morris) [reply bsp/esp_bsp_generic/src/esp_bsp_generic.c:198]: leave this field empty means, i.e. init the field to `0`, the led_strip will fall-back to use the default format "GRB" as well.

Then this esp_bsp_generic package can work both with led_strip version2 and led_strip version3.

**Comment by suda-morris** [comment]
> cc @espzav @tore-espressif
> 
> I think we also need to add a version constraint on led_strip somewhere, or possibly have ifdefs for led_strip v2 and v3.

We'd better to set a proper dependency in [here](https://github.com/espressif/esp-iot-solution/blob/master/components/led/led_indicator/idf_component.yml#L7)

**Comment by igrr** [comment]
True, led_indicator component should also change the way the dependency is set.

However, since esp_bsp_generic also uses led_strip API directly, it should specify the dependency and not rely on the version set in led_indicator.

Besides, I think it's better to allow both versions of led_strip, if possible. If we force led_strip>=3, we have to make a new major release of the BSP component. If we force led_strip<3, users won't be able to use esp_bsp_generic with led_strip>=3.

**Comment by suda-morris** [comment]
> Besides, I think it's better to allow both versions of led_strip, if possible.

Yes, we can. Just remove the configuration of `led_pixel_format` and `color_component_format` when initialize the led_strip, let the driver fall-back to choose the default one. The esp_bsp_generic can compile with led_strip version 2 and 3.

**Comment by suda-morris** [comment]
I take the liberty to fix the build errors that are introduced by led_strip v3. Now the bsp package can work with both led_strip v2 and led_strip v3. @igrr @espzav PTAL.

**Review by suda-morris — APPROVED** [review]

**Review by Lzw655 — APPROVED** [review]
Hi @suda-morris, LGTM. Please also update the version of the `esp32-c3-lcdkit` BSP. I tested the LED functionality using the compatible development board, and everything works fine.

**Comment by suda-morris** [comment]
> Hi @suda-morris, LGTM. Please also update the version of the `esp32-c3-lcdkit` BSP. I tested the LED functionality using the compatible development board, and everything works fine.

c3-lcdkit already has a contraint to the led_strip:

```yml
  led_strip:
    version: "^2"
    public: true
```

So this PR actually didn't do any bugfix for that BSP, it's just a refactor.

**Pull request merged**